src/main/java/io/supertokens/webserver/api/dashboard/DashboardUserAPI.java

```
@@ -0,0 +1,207 @@
+            response.addProperty("status", "EMAIL_ALREADY_EXISTS_ERROR");
+            super.sendJsonResponse(200, response, resp);
+
+        } catch (StorageQueryException | StorageTransactionLogicException e) {
+            throw new ServletException(e);
+        }
+
+        JsonObject response = new JsonObject();
+        response.addProperty("status", "OK");
+        super.sendJsonResponse(200, response, resp);
+    }
+
+    @Override
+    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+
+        String userId = InputParser.parseStringOrThrowError(input, "userId", true);
+        try {
+            if (userId != null) {
+                // normalize userId
+                userId = normalizeStringParam(userId, "userId");
+                boolean didUserExist = Dashboard.deleteUserWithUserId(main, userId);
+                JsonObject response = new JsonObject();
+                response.addProperty("status", "OK");
+                response.addProperty("didUserExist", didUserExist);
+                super.sendJsonResponse(200, response, resp);
+                return;
+            }
+
+            String email = InputParser.parseStringOrThrowError(input, "email", true);
+
+            if (email != null) {
+                // normalize email
+                email = normalizeStringParam(email, "email");
+                boolean didUserExist = Dashboard.deleteUserWithEmail(main, email);
+                JsonObject response = new JsonObject();
+                response.addProperty("status", "OK");
+                response.addProperty("didUserExist", didUserExist);
+                super.sendJsonResponse(200, response, resp);
+                return;
+            }
+
+        } catch (StorageQueryException e) {
+            throw new ServletException(e);
+        }
+
+        JsonObject response = new JsonObject();
```

The API should throw a 400 if both userId and email are missing